src/mycollections/DoubleLinkedList.java

```diff
@@ -1,59 +1,245 @@
 package mycollections;
 
-public class DoubleLinkedList<T> implements MyList<T>{
+public class DoubleLinkedList<T> implements MyList<T> {
+
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
 
     @Override
     public int size() {
-        return 0;
+        return this.size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
     }
 
     @Override
     public boolean contains(T o) {
-        return false;
+        elementValueCheck(o);
+        return indexOf(o) != -1;
     }
 
     @Override
     public int indexOf(T o) {
-        return 0;
+        elementValueCheck(o);
+        int index = 0;
+        for (Node<T> first = head; first != null; first = first.next) {
+            if (first.item.equals(o)) {
+                return index;
+            }
+            index++;
+        }
+        return -1;
     }
 
     @Override
     public int lastIndexOf(T o) {
-        return 0;
+        elementValueCheck(o);
+        int index = 0;
+        for (Node<T> last = tail; last != null; last = last.prev) {
+            if (last.item.equals(o)) {
+                return index;
+            }
+            index++;
+        }
+        return -1;
     }
 
     @Override
     public T get(int index) {
+        rangeCheck(index);
+        if (size != 0) {
+            int halfLength = (size >> 1);
+            if (index <= halfLength && head != null) {
+                for (int i = 0; i <= halfLength; i++) {
+                    if (i == index) {
+                        return head.item;
+                    }
+                    head = head.next;
+                }
+            } else if (index > halfLength && tail != null) {
+                for (int i = size; i > halfLength; i--) {
+                    if (i == index) {
+                        return tail.item;
+                    }
+                    tail = tail.prev;
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public T set(int index, T element) {
```

Comment:
This method is working incorrectly please debug it. Here you also have next and prev's linking logic errors.